prboom2/src/r_bsp.c

```diff
@@ -619,7 +631,7 @@ static void R_Subsector(int num)
   sub = &subsectors[num];
   currentsubsectornum = num;
 
-  if (V_IsSoftwareMode() || !gl_use_stencil || sub->sector->validcount != validcount)
+  if (V_IsSoftwareMode() || sub->sector->gl_validcount != validcount)
```

Comment:
So what exactly is going on in this function 🤔 
Why does gl only want to do this once, but for software it happens each time? Or does software only reach this once per sector?

Reply:
I think this could be changed to only do it once per sector for software mode as well, but I'd have to test it.  It's basically doing a bunch of hash lookups for no reason after the first subsector in a sector.

Reply:
Those plane globals need to be updated any time the sector changes, and the order subsectors are processed is probably not guaranteed to be contiguous by sector, so that may be why it has to be redone.

Reply:
But actually, wouldn't that be the same for gl then? 🤔

Reply:
Ah, right.  `floorplane` and `ceilingplane` are always needed for software because they get used by `R_StoreWallRange` in `r_segs.c`, but GL only needs them when it could potentially call `gld_AddPlane`.  So this can be left as-is.